verilog/tools/kythe/kythe_facts_extractor.cc

```diff
@@ -696,18 +783,19 @@ const std::set<Edge>& KytheFactsExtractor::GetExtractedEdges() const {
   return edges_;
 }
 
-std::string GetFilePathFromRoot(const IndexingFactNode& root) {
-  return root.Value().Anchors()[0].Value();
-}
-
 std::ostream& KytheFactsPrinter::Print(std::ostream& stream) const {
-  auto indexing_data = CollectKytheIndexingData(trees_);
-  for (const Fact& fact : indexing_data.facts) {
+  ScopeResolver scope_resolver(nullptr);
+  KytheFactsExtractor kythe_extractor(
+      GetFileListDirFromRoot(file_list_facts_tree_), &scope_resolver);
```

Comment:
I'm still a little concerned that printing requires passing a mutable object to the extractor, but perhaps I need to wrap my head around everything that is happening.  Can you add some comments about this? or point me to comments if you already have?